smslibrary/src/test/java/com/eis/smslibrary/SMSHandlerTest.java

```diff
@@ -0,0 +1,68 @@
+package com.eis.smslibrary;
+
+import android.content.Context;
+import android.content.IntentFilter;
+import android.telephony.SmsManager;
+
+import com.eis.smslibrary.listeners.SMSSentListener;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SMSHandlerTest {
+
+    @Mock SmsManager managerMock;
+    @Mock Context contextMock;
+
+    private final String PEER_TEXT = "+393423541601";
+    private final String MESSAGE_TEXT = "valid text";
+    private final String SENT_TEXT = (char)0x02 + MESSAGE_TEXT;
+    private final SMSPeer VALID_PEER = new SMSPeer(PEER_TEXT);
+    private final SMSMessage VALID_MESSAGE = new SMSMessage(VALID_PEER, MESSAGE_TEXT);
+    private SMSHandler instance;
+
+    @Before
+    public void setup(){
+        instance = SMSHandler.getInstance();
+        instance.clearSetup();
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void sendMessageWithNullContext_throws(){
+        instance.sendMessage(VALID_MESSAGE);
+    }
+
+    @Test
+    public void singletonInstance(){
+        assertEquals(SMSHandler.getInstance(), SMSHandler.getInstance());
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void setupNullContext_throws(){
+        instance.setup(null);
+    }
+
+    @Test()
+    public void validMessage_isSent(){
+        SMSCore.setManager(managerMock);
```

Comment:
Instead of writing a method in `SMSCore` that will only be used in tests, couldn't you just stub the behavior of `SMSCore` for method `getManager()` to make it return your mocked `SmsManager`?

Reply:
1. SMSCore is only static methods, so I can't use Mock objects (except PowerMock that I didn't know at the time of writing this, and that I don't think I'm gonna use because of number 2)
2. Here I'm trading 2 simple methods used only for tests, for an entire class tested, I think this is a good compromise to have: 2 simple basic methods vs an entire class tested.

Reply:
> Here I'm trading 2 simple methods used only for tests, for an entire class tested, I think this is a good compromise to have: 2 simple basic methods vs an entire class tested.

What do you mean with this? Using `PowerMock` would allow you to remove one of those 2 methods and still be able to test the class, it seems like the best compromise.

Reply:
At the time of writing that code I didn't know about PowerMock, and I still have to check if it's reasonable to use it. If it is I'll definitely change this to using power mock, but if it becomes too tedious or impossible I'll keep this 2 methods.
What I meant with that sentence is that thanks to those 2 simple methods (which cannot be used outside of the package btw) i'm able to create unit-tests for an entire class, which in my opinion is a reasonable compromise to do.